common/lib/xmodule/xmodule/tests/test_course_metadata_utils.py

```diff
@@ -0,0 +1,204 @@
+            ]),
+            FunctionTest(number_for_course_location, [
+                TestScenario((self.demo_course.location,), "DemoX.1"),
+                TestScenario((self.html_course.location,), "CS-203"),
+            ]),
+            FunctionTest(has_course_started, [
+                TestScenario((self.demo_course.start,), True),
+                TestScenario((self.html_course.start,), False),
+            ]),
+            FunctionTest(has_course_ended, [
+                TestScenario((self.demo_course.end,), True),
+                TestScenario((self.html_course.end,), False),
+            ]),
+            FunctionTest(course_start_date_is_default, [
+                TestScenario((test_datetime, advertised_start_parsable), False),
+                TestScenario((test_datetime, None), False),
+                TestScenario((DEFAULT_START_DATE, advertised_start_parsable), False),
+                TestScenario((DEFAULT_START_DATE, None), True),
+            ]),
+            FunctionTest(course_start_datetime_text, [
+                TestScenario(
+                    (DEFAULT_START_DATE, advertised_start_parsable, 'DATE_TIME', ugettext, mock_strftime_localized),
+                    mock_strftime_localized(Date().from_json(advertised_start_parsable), 'DATE_TIME') + " UTC"
+                ),
+                TestScenario(
+                    (test_datetime, advertised_start_unparsable, 'DATE_TIME', ugettext, mock_strftime_localized),
+                    advertised_start_unparsable.title()
+                ),
+                TestScenario(
+                    (test_datetime, None, 'SHORT_DATE', ugettext, mock_strftime_localized),
+                    mock_strftime_localized(test_datetime, 'SHORT_DATE')
+                ),
+                TestScenario(
+                    (DEFAULT_START_DATE, None, 'SHORT_DATE', ugettext, mock_strftime_localized),
+                    # Translators: TBD stands for 'To Be Determined' and is used when a course
+                    # does not yet have an announced start date.
+                    ugettext('TBD')
+                )
+            ]),
+            FunctionTest(course_end_datetime_text, [
+                TestScenario(
+                    (test_datetime, 'TIME', mock_strftime_localized),
+                    mock_strftime_localized(test_datetime, 'TIME') + " UTC"
+                ),
+                TestScenario(
+                    (None, 'TIME', mock_strftime_localized),
+                    ""
+                )
+            ]),
+            FunctionTest(may_certify_for_course, [
+                TestScenario(('early_with_info', True, True), True),
+                TestScenario(('early_no_info', False, False), True),
+                TestScenario(('end', True, False), True),
+                TestScenario(('end', False, True), True),
+                TestScenario(('end', False, False), False),
+            ]),
+        ]
+
+        debug_print_test_output = False
```

Comment:
Hmm.. I don't necessarily like checking in code that has no test coverage and is not regularly run.  Is it necessary to add this debug code that is never executed?